### PR TITLE
feat(js): add session id parsing to langchain instrumentation

### PR DESCRIPTION
**Pull request by cephalization**
Parses metadata with key name in (session_id, thread_id, conversation_id) into semantic conventions session id

### CODE REVIEW

File: js/packages/openinference-instrumentation-langchain/examples/chat.ts [hunk 1]
```diff
@@ -1,18 +1,31 @@
...
+    response,
+    new HumanMessage("That is great to hear!"),
+  ]);
 
   // eslint-disable-next-line no-console
-  console.log(response.content);
+  console.log(finalResponse.content);
 
-  return response;
+  return finalResponse;
```
Comment (cephalization) [inline js/packages/openinference-instrumentation-langchain/examples/chat.ts:28]: Happy to revert these changes if they seem too complex for generic example code

File: js/packages/openinference-instrumentation-langchain/src/utils.ts [hunk 2]
```diff
@@ -650,6 +650,32 @@ function formatMetadata(run: Run) {
...
+/**
+ * Formats the session id of a langchain run into OpenInference attributes.
+ *
+ * @see https://docs.smith.langchain.com/observability/how_to_guides/monitoring/threads#group-traces-into-threads
+ *
+ * @param run - The langchain run to extract the session id from
+ * @returns The OpenInference attributes for the session id
+ */
+function formatSessionId(run: Run) {
+  const sessionKeys = ["session_id", "thread_id", "conversation_id"];
```
Comment (mikeldking) [inline js/packages/openinference-instrumentation-langchain/src/utils.ts:662]: Can be hoisted out into a constant

**Review by mikeldking — APPROVED** [review]

File: js/packages/openinference-instrumentation-langchain/src/utils.ts (hunk 2 above)
Comment (cephalization) [reply js/packages/openinference-instrumentation-langchain/src/utils.ts]: Done
